logictest: remove explicit set_vmodule call in index_join

asubiotto opened the pull request:
The nightly high-verbosity logic tests already take care of testing this.

Release note: None (testing change)

Addresses #54839

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55021)
<!-- Reviewable:end -->

rytaft reviewed (approved):
Nice catch! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-:-MITmm-T0mzO6LdxagN7:bppivfc)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto and @yuzefovich)

___
*[pkg/sql/logictest/testdata/logic_test/index_join, line 44 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-MITmjFKE03bEiM5RMl6:-MITmjFKE03bEiM5RMl7:bccpm19) ([raw file](https://github.com/cockroachdb/cockroach/blob/4b998e990832c42dbac4103e58d4d694a18225a8/pkg/sql/logictest/testdata/logic_test/index_join#L44)):*
> ```
> # all subsequent tests.
> statement ok
> SELECT crdb_internal.set_vmodule('processorsbase=0')
> ```

is this the default value?


<!-- Sent from Reviewable.io -->

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55021)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/sql/logictest/testdata/logic_test/index_join, line 44 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-MITmjFKE03bEiM5RMl6:-MITpEXB6188EIOMDWBg:b-ci8wnu) ([raw file](https://github.com/cockroachdb/cockroach/blob/4b998e990832c42dbac4103e58d4d694a18225a8/pkg/sql/logictest/testdata/logic_test/index_join#L44)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

is this the default value?
</blockquote></details>

This is a good point. It's not the default value because we also run logic tests with a high verbosity just to make sure all the high-verbosity logging doesn't produce any panics. @yuzefovich I think we should just remove setting a vmodule explicitly since the high verbosity tests should do this for us.


<!-- Sent from Reviewable.io -->

asubiotto reviewed (commented):
Hmm, I actually don't think this has any effect. For some reason I thought we reused clusters in logic tests but this is not the case.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-:-MIU8PR6DmscqhILSt3_:bloblzh)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
Wait, I thought so too - are you saying we're spinning up a new test cluster for each file?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-:-MIUX6Fp4C42SZfkP00b:b-slvrgk)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rytaft)

___
*[pkg/sql/logictest/testdata/logic_test/index_join, line 44 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-MITmjFKE03bEiM5RMl6:-MIUWtYO82x-1nxBXoQO:bbg9i5p) ([raw file](https://github.com/cockroachdb/cockroach/blob/4b998e990832c42dbac4103e58d4d694a18225a8/pkg/sql/logictest/testdata/logic_test/index_join#L44)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

This is a good point. It's not the default value because we also run logic tests with a high verbosity just to make sure all the high-verbosity logging doesn't produce any panics. @yuzefovich I think we should just remove setting a vmodule explicitly since the high verbosity tests should do this for us.
</blockquote></details>

I'm in favor of removing the setting - I added it before high vmodule tests were working. 


<!-- Sent from Reviewable.io -->

Commit:
logictest: remove explicit set_vmodule call in index_join

The nightly high-verbosity logic tests already take care of testing this.

Release note: None (testing change)

asubiotto commented:
Updated.

yuzefovich reviewed (approved):
Indeed, you're right that we don't reuse the test clusters between different tests, so I'm not sure whether this change makes a difference, but it's a good idea to update this test anyway. <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-:-MIUkAF64pR88k9iiixG:bii1037)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @rytaft)



<!-- Sent from Reviewable.io -->

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55021#-:-MIUqT-J1eDr0KGAbBPZ:bg7igkv)* status: :shipit: complete! 2 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

asubiotto commented:
TFTRs!

bors r=rytaft,yuzefovich
